composeApp/src/commonMain/kotlin/sp/bvantur/tasky/login/domain/LoginUseCase.kt

```diff
@@ -0,0 +1,19 @@
+package sp.bvantur.tasky.login.domain
+
+import sp.bvantur.tasky.login.data.LoginRepository
+
+class LoginUseCase(private val repository: LoginRepository) {
+    suspend operator fun invoke(email: String, password: String): Boolean {
+        val response = repository.login(email, password)
+        if (response.isSuccess) {
+            val userInformation = response.getOrNull() ?: return false
+
+            repository.storeTokenInformation(
+                userInformation.accessToken,
+                userInformation.accessTokenExpirationTimestamp
+            )
+            return true
+        }
+        return false
+    }
```

Comment:
All this use cases does is to delegate calls to your repository. Why can't the repository do that on its own already? After all, the job of a repository is to bundle multiple data sources and connect them. In this case, it would be the remote data source and the local token storage, so the repo could already insert the token there after a successful call to login. The use case wouldn't be necessary then.

Reply:
@philipplackner So if I understand this correctly, the repository is responsible for everything data-related? In this case, it should store an access token. In a case if we are using a JWT token, again repository would be responsible for parsing the token and then storing all of the data to appropriate data sources. Am I understanding this correctly?

Reply:
Yes, a repository is a design pattern to bundle multiple data sources, like to define a caching/syncing strategy or to save tokens after logging in. If a class just serves as a source of data, I'd call it `XxxDataSource`